### ISSUE TITLE
find terraria path properly

### ISSUE THREAD

**Issue by flying-sheep** [Berserker66/omnitool#12]
once #11 is fixed, this is easy:

mac/linux: `appdirs.user_data_dir('Terraria')`
windows; unchanged: `Documents/My Games/Terraria`

but worlds are also in `os.path.join(appdirs.user_data_dir('Steam'), 'steam', 'userdata', user_id, '105600', 'remote', 'worlds')`

**Comment by Berserker66** [comment]
offline world paths adjusted in cbeb3025c10d6216cac57b22fdeb02e6180359cf .
Note that this breaks compatibility with old monogame modded Terraria versions, which saved to ~/My Games/Terraria

**Comment by Berserker66** [comment]
As for the online world it may be required to sweep the entire dir tree, since I don't think it's easily possible to determine currently logged in steam user id.

**Comment by flying-sheep** [comment]
wow, thanks, great!

btw., caches are usually in `user_cache_dir`, but that’s not important. i’m leaving for the festival in some minutes, at sunday i’ll be able to contribute :smile:

**Comment by Berserker66** [comment]
it's not really a cache. it's settings and meta-info on processed worlds. Which in future might be useful to split into separate files.

**Comment by Berserker66** [comment]
Current implementation of getting Terraria online worlds broke somewhere along the line.
I fixed it for ubuntu in fe42b24ac8690efc9e43d4c6c0e077e630fa1f37
If that doesn't work in other linux' we'll need to start checking for multiple locations or react based on OS.

**Comment by flying-sheep** [comment]
should work, but idk why you did the first change.

`appdirs.user_data_dir('Steam')`, i.e. `~/.local/share/Steam` should be symlinked to by `~/.steam/steam`

**Comment by Berserker66** [comment]
That's not how my ubuntu handled it - appdirs.user_data_dir('Steam') does exist, but only has a logs folder, nothing else.

**Comment by Berserker66** [comment]
Right, it appears to work. I'll close this last issue of the V18 milestone and start deploying.
Linux binaries don't exist yet, but there is no reason to have the windows users wait for that.
Will continue to work on linux binaries over the lifetime of V18. I tried Pyinstaller as freezer so far, but that does not handle any "from omnitool import ..." at all, and there's plenty of those. Admittedly an import from `__main__` is bad practice - which I did not know 3 years ago when that was first written ;)
